### ISSUE TITLE
join and delete ex commands don't take count argument

### ISSUE THREAD

**Issue by quaker4lyf** [emacs-evil/evil#760]
Example lines in my file:
```
pattern
  a
  b

pattern
  c
  d

pattern
  e
  f
```
`:%g/pattern/j 3` in vim it produces what I want:
```
pattern a b

pattern c d

pattern e f
```
whereas in spacemacs (using evil) it only joins 2 lines (which is the same as ignoring the count):
```
pattern a
  b

pattern c
  d

pattern e
  f
```
Similar thing with the `:delete` command.
`:%g/pattern/d 3` in vim leaves only the empty lines:
```


```
whereas in spacemacs it deletes only the lines matching the pattern
```
  a
  b

  c
  d

  e
  f
```
I figure the bug has to do with evil hence filing it here.

Emacs version: GNU Emacs 25.1.91.1
Spacemacs version: spacemacs 0.200.5@25.1.91.1
(Please let me know if you need the Evil version)

**Comment by edkolev** [comment]
Since #818 is now merged, this issue can be closed.

**Comment by wasamasa** [comment]
Good call!  BTW, putting a `Fixes #818` or `Closes #818` line into the commit message causes the linked issue to be closed as soon as the MR gets merged.

**Comment by edkolev** [comment]
I see, thanks!
